provider/src/main/java/com/raygun/raygun4android/RaygunClient.java

```diff
@@ -55,6 +56,43 @@ public static void init(Application application, String apiKey) {
         init(application, apiKey, null);
     }
 
+    /**
+     * Initializes the Raygun client with applicationContext and your Raygun API key. The version
+     * transmitted will be the value of the versionName attribute in your manifest element. This
+     * function should be used by 3rd party library.
+     *
+     * @param Context The Android applicationContext
+     * @param apiKey An API key that belongs to a Raygun application created in your dashboard
+     */
+    public static void init(Context applicationContext, String apiKey) {
```

Comment:
Your new code has a lot of duplication with 

public static void init(Application application, String apiKey, String version) { ... }

I'd rather have that code exist only once and both the new constructor for RN usage as well as the existing one fall through to the same code (apart from assigning RGCLient.application resp. .applicationContext)